### PR TITLE
Clean up on failed BIO creation

### PR DESCRIPTION
**Pull request by bonniegong**
Check the result of BIO_new and clean up when fail

CLA: trivial

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by t8m — COMMENTED** [review]
This is too significant change for being acceptable with CLA: trivial. Could you please submit a regular CLA?

**Comment by bonniegong** [comment]
> This is too significant change for being acceptable with CLA: trivial. Could you please submit a regular CLA?

Thank you for informing. I have submitted a CLA just now.

File: apps/s_client.c [hunk 1]
```diff
@@ -2134,7 +2136,8 @@ int s_client_main(int argc, char **argv)
              */
             int foundit = 0;
             BIO *fbio = BIO_new(BIO_f_buffer());
-
+            if (fbio == NULL)
```
Comment (paulidale) [inline apps/s_client.c:2139]: Blank line before.

File: apps/s_client.c [hunk 2]
```diff
@@ -2182,7 +2185,8 @@ int s_client_main(int argc, char **argv)
         {
             int foundit = 0;
             BIO *fbio = BIO_new(BIO_f_buffer());
-
+            if (fbio == NULL)
```
Comment (paulidale) [inline apps/s_client.c:2188]: Blank line before.

File: apps/s_client.c [hunk 3]
```diff
@@ -2209,7 +2213,8 @@ int s_client_main(int argc, char **argv)
     case PROTO_FTP:
         {
             BIO *fbio = BIO_new(BIO_f_buffer());
-
+            if (fbio == NULL)
```
Comment (paulidale) [inline apps/s_client.c:2216]: Blank line before.

File: apps/s_client.c [hunk 4]
```diff
@@ -2303,7 +2308,8 @@ int s_client_main(int argc, char **argv)
         {
             int numeric;
             BIO *fbio = BIO_new(BIO_f_buffer());
-
+            if(fbio == NULL)
```
Comment (paulidale) [inline apps/s_client.c:2311]: Blank line before.
Space between `if` and `(`.

File: apps/s_client.c [hunk 5]
```diff
@@ -2463,7 +2469,8 @@ int s_client_main(int argc, char **argv)
         {
             int foundit = 0;
             BIO *fbio = BIO_new(BIO_f_buffer());
-
+            if (fbio==NULL)
```
Comment (paulidale) [inline apps/s_client.c:2472]: Blank line before.
Spaces either side of the `==`.

File: apps/s_client.c [hunk 6]
```diff
@@ -2503,7 +2510,8 @@ int s_client_main(int argc, char **argv)
...
```
Comment (paulidale) [inline apps/s_client.c:2513]: Blank line before.
Spaces either side of the `==`.

File: apps/s_server.c [hunk 7]
```diff
@@ -3400,7 +3408,7 @@ static int rev_body(int s, int stype, int prot, unsigned char *context)
     int ret = 1;
     SSL *con;
     BIO *io, *ssl_bio, *sbio;
-
+    BIO *BIO = NULL
```
Comment (paulidale) [inline apps/s_server.c:3411]: Blank line after.

**Review by paulidale — COMMENTED** [review]
Just code style nit picks.

File: apps/s_server.c [hunk 8]
```diff
@@ -3029,7 +3034,10 @@ static int www_body(int s, int stype, int prot, unsigned char *context)
     BIO_push(io, ssl_bio);
     ssl_bio = NULL;
 #ifdef CHARSET_EBCDIC
-    io = BIO_push(BIO_new(BIO_f_ebcdic_filter()), io);
+    bio = BIO_new(BIO_f_ebcdic_filter());
+    if (bio == NULL)
+        goto err;
+    io = NIO_push(bio, io);
```
Comment (t8m) [inline apps/s_server.c:3040]: Typo: NIO_push

File: apps/s_server.c [hunk 9]
```diff
@@ -2972,6 +2974,7 @@ static int www_body(int s, int stype, int prot, unsigned char *context)
     SSL *con;
     const SSL_CIPHER *c;
     BIO *io, *ssl_bio, *sbio;
+    BIO *bio = NULL;
```
Comment (t8m) [inline apps/s_server.c:2977]: Please enclose this in `#ifdef CHARSET_EBCDIC` otherwise we get the compilation failure with  -Werror.

**Review by t8m — CHANGES_REQUESTED** [review]

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
The CI failures are relevant.

File: apps/s_server.c [hunk 10]
```diff
@@ -3400,6 +3410,7 @@ static int rev_body(int s, int stype, int prot, unsigned char *context)
     int ret = 1;
     SSL *con;
     BIO *io, *ssl_bio, *sbio;
+    BIO *bio = NULL;
```
Comment (t8m) [inline apps/s_server.c:3413]: This has to be again surrounded with `#ifdef CHARSET_EBCDIC`

**Comment by beldmit** [comment]
Test failures are relevant

**Pull request closed without merging**